File: src/main/api/site.ts
interface Config {
  baseUrl: string
  userAgent: string
  test?: boolean
}

export class SiteManager {
  private config
  constructor(config: Config) {
    this.config = config
  }
  getSiteInfo(): string {
    return `Fetching site info from ${this.config.baseUrl}`
  }
}

File: src-tauri/src/api/site.rs
use crate::api::config::Config;

#[derive(Clone)]
pub struct SiteManager {
    config: Config,
    client: reqwest::Client,
}

// Работа с сайтом, скачивание, получение, всякие вот эти, файлы, остальное
impl SiteManager {
    pub fn new(config: &Config, client: reqwest::Client) -> Self {
        Self {
            config: config.clone(),
            client,
        }
    }

    pub fn get_site_info(&self) -> String {
        format!("Fetching site info from {}", self.config.base_url)
    }
}
